src/hotspot/share/opto/superword.cpp

```diff
@@ -624,44 +576,6 @@ bool SuperWord::SLP_extract() {
     DEBUG_ONLY(verify_packs();)
 
     schedule();
-
-    // Record eventual count of vector packs for checks in post loop vectorization
-    if (PostLoopMultiversioning) {
-      cl->set_slp_pack_count(_packset.length());
-    }
-  } else {
```

Comment:
Again: Could we now have an assert for `cl->is_main_loop()` at the beginning of `SuperWord::SLP_extract`, and remove all checks for it in SuperWord?

Reply:
Ok to do it here as `do_optimization` is false in the unrolling analysis phase. I've updated the code in commit 2.